src/workerd/api/container.c++

```diff
@@ -56,14 +65,21 @@ jsg::Promise<void> Container::monitor(jsg::Lock& js) {
   JSG_REQUIRE(running, Error, "monitor() cannot be called on a container that is not running.");
 
   return IoContext::current()
-      .awaitIo(js, rpcClient->monitorRequest(capnp::MessageSize{4, 0}).sendIgnoringResult())
-      .then(js, [this](jsg::Lock& js) {
+      .awaitIo(js, rpcClient->monitorRequest(capnp::MessageSize{4, 0}).send())
+      .then(js, [this](jsg::Lock& js, capnp::Response<rpc::Container::MonitorResults> results) {
     running = false;
+    auto exitCode = results.getExitCode();
     KJ_IF_SOME(d, destroyReason) {
       jsg::Value error = kj::mv(d);
       destroyReason = kj::none;
       js.throwException(kj::mv(error));
     }
+
+    if (exitCode != 0) {
+      auto err = js.error(kj::str("Container exited with unexpected exit code: ", exitCode));
+      setExitCodeError(js, err, exitCode);
```

Comment:
I think this can be replaced with:

```suggestion
      KJ_ASSERT_NONNULL(err.tryCast<jsg::JsObject>()).set("exitCode", js.num(exitCode));
```

(I'm pretty sure `jsg.error()` _always_ returns an object, but unfortunately the underlying V8 APIs are declared only as returning `Value`.)